openapi.yaml

```diff
@@ -0,0 +1,2069 @@
+      description: The collection does not contain data. To get data, you must get a single entity.
+      tags: [Variable]
+      responses:
+        '200':
+          description: Successful response.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Variable'
+        '401':
+          $ref: '#/components/responses/Unauthenticated'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+        '404':
+          $ref: '#/components/responses/NotFound'
+
+    patch:
+      summary: Update a variable by id
+      operationId: updateVariable
+      tags: [Variable]
+      parameters:
+        - $ref: '#/components/parameters/UpdateMask'
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Variable'
+
+      responses:
+        '200':
+          description: Successful response.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Variable'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthenticated'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+        '404':
+          $ref: '#/components/responses/NotFound'
+
+    delete:
+      summary: Delete variable
+      operationId: deleteVariable
+      tags: [Variable]
+      responses:
+        '204':
+          description: No content.
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthenticated'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+
+  /variables/lookup:
+    parameters:
+      - $ref: '#/components/parameters/VariableKey'
+
+    get:
+      summary: Get variable using its key.
+      description: The response contains all fields, including data.
+      operationId: lookupVariable
+      tags: [Variable]
+      responses:
+        '200':
+          description: Successful response.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Variable'
+        '401':
+          $ref: '#/components/responses/Unauthenticated'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+        '404':
+          $ref: '#/components/responses/NotFound'
+
+  /dags/{dag_id}/taskInstances/{task_id}/{execution_date}/xcomValues:
+    parameters:
+      - $ref: '#/components/parameters/DAGID'
+      - $ref: '#/components/parameters/TaskID'
+      - $ref: '#/components/parameters/ExecutionDate'
+
+    get:
+      summary: Get all XCom values
+      description:
+        This endpoint support reading resources across multiple Task Instances by specifying a
+        "-" as a `dag_id`, `task_id` and `execution_date`.
+      operationId: getXComValues
```

Comment:
Because the value [is/should be] just one field of the returned by XcomCollectionItem these operationIds under this path which end in Value(s) get a little confusing. Maybe we could rename them like: `getXComEntries`? The same can be said about the summary and description mentioning just the values.

Reply:
I updated spec. Here is commit: https://github.com/apache/airflow/pull/8721/commits/26e0d2292e00374623d6c72c869bbd8bf57d0ab3